noloco/utils.py

```diff
@@ -131,25 +133,42 @@ def find_relationship_data_type(
         # If the relationship field exists on the parent data type then
         # this is a forward relationship and we can simply look up the
         # relationship data type by the corresponding field type.
-        return find_data_type_by_name(relationship_field['type'], data_types)
+        return {
+            'data_type': find_data_type_by_name(
+                relationship_field['type'], data_types),
+            'is_collection': relationship_field['relationship'] == MANY_TO_MANY
+        }
     else:
         # If there isn't a corresponding relationship field on the
         # parent data type then this is a reverse relationship and we
-        # have to search for the relationship data type by it having a
-        # field of the expected reverseName and type matching the
-        # parent type.
+        # have to search for the relationship data type.
         for candidate_data_type in data_types:
-            candidate_fields = [
-                field
-                for field
-                in candidate_data_type['fields']
-                if field['reverseName'] is not None]
-
-            for field in candidate_fields:
-                reverseName = field['reverseName'] + 'Collection'
-                if field['type'] == data_type_name and \
-                        reverseName == relationship_name:
-                    return candidate_data_type
+            for field in candidate_data_type['fields']:
+                # We are looking for any fields whose type matches up with the
+                # name of the parent data type that the reverse relationship is
+                # from.
+                if field['type'] == data_type_name:
+                    # If the reverse name is populated and matches the parent
+                    # type then we have found the data type the relationship
+                    # goes to.
+                    if field['reverseName'] is not None and \
+                            field['reverseName'] != '':
+                        reverseName = field['reverseName'] + 'Collection'
+                        if reverseName == relationship_name:
+                            return {
+                                'data_type': candidate_data_type,
+                                'is_collection': True
+                            }
+                    # If the reverse name is not populated but this field is a
+                    # one-to-one relationship and the data type name matches
+                    # the relationship name then we have found the data type
+                    # the relationship goes to.
+                    elif field['relationship'] == ONE_TO_ONE:
```

Comment:
As mentioned above, you're omitting the reverse of `ONE_TO_MANY` which is singular, but uses `reverseName` as a key